File: 백준/Gold/1931. 회의실 배정/회의실 배정.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());

		int[][] time = new int[n][2];

		for(int i=0; i<n; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			int start = Integer.parseInt(st.nextToken());
			int end = Integer.parseInt(st.nextToken());

			time[i][0] = start;
			time[i][1] = end;
		}

		Arrays.sort(time, (o1, o2) -> {
			if (o1[1] == o2[1]) {
				return o1[0] - o2[0];
			}return o1[1] - o2[1];
		});

		int numRoom = 1;
		int endTime = time[0][1];

		for(int i=1; i<n; i++) {
			if (time[i][0] >= endTime) {
				numRoom++;
				endTime = time[i][1];
			}
		}

		System.out.println(numRoom);


    }
}

File: 백준/Gold/1931. 회의실 배정/회의실 배정.py
import sys 
N = int(sys.stdin.readline()) 
time = [[0]*2 for _ in range(N)] 
for i in range(N): 
    s, e = map(int, sys.stdin.readline().split())
    time[i][0] = s 
    time[i][1] = e
time.sort(key = lambda x: (x[1], x[0]))
numRoom = 1
endTime = time[0][1]

for i in range(1,N):
	if time[i][0] >= endTime:
		numRoom += 1
		endTime = time[i][1]
print(numRoom)